### ISSUE TITLE
`isStandardSyntaxSelector` reads a Less guard keyword in any case, where Less reads it in lower case only

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#168]
Found while fixing [#150](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/150), which asks the Less compiler what a guard may look like. One of the answers contradicts an assumption [#125](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/125) built in: `when` is a keyword, and Less reads keywords in lower case only.

### What minimal example or steps are needed to reproduce the bug?

```less
.a:hover WHEN (1 = 1), .b { color: red; }
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "customSyntax": "postcss-less",
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/selector-list-comma-newline-after": "always"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.less" --config myconfig.json --fix
```

### Which Stylelint-related dependencies are you using?

```json
{
  "@stylistic/stylelint-plugin": "5.3.0",
  "stylelint": "17.14.1"
}
```

### What did you expect to happen?

A warning about the comma, and a fix writing the line break the option asks for — since Less does not read this as a guard.

### What actually happened?

Nothing is reported, and nothing is fixed. `isStandardSyntaxSelector` takes the selector for a Less construct, so every rule built on it — `no-extra-semicolons`, `selector-attribute-brackets-space-inside`, `selector-descendant-combinator-no-non-space`, `selector-list-comma-newline-after`, `selector-pseudo-class-case`, `selector-pseudo-class-parentheses-space-inside`, `selector-pseudo-element-case` and `string-quotes` — passes the rule node over.

The compiler is plain about it. `less@4` emits all three of these verbatim, guard and all, because it never read a guard there:

```less
.a:hover WHEN (1 = 1) { color: red; }   →   .a:hover WHEN (1 = 1) { color: red; }
.a:hover When (1 = 1) { color: red; }   →   .a:hover When (1 = 1) { color: red; }
.a:hover WHEN(1 = 1) { color: red; }    →   .a:hover WHEN(1 = 1) { color: red; }
```

while the lower-case forms are read as guards and compile to `.a:hover { color: red; }`. `not` is the same: `when NOT (1 = 1)` is a hard `SyntaxError: expected condition`, where `when not (1 = 1)` drops the rule as a false guard does.

### Do you have a proposal to fix the bug?

The test is

```js
if ((/\swhen\s*(?:not\s*)?\(/iu).test(withoutQuotedText(selector))) return false
```

and the `i` flag is what asks for this. Taking it away is the whole of the change, along with the two cases in `lib/utils/isStandardSyntaxSelector/index.test.js` that assert the opposite (`a guard written in another case`) and the accept cases that came with them.

Nothing valid is lost either way, which is why this is a small thing: `.a:hover WHEN (1 = 1)` is not CSS a browser would read, since a parenthesis opening after whitespace is nothing a selector has. What changes is what the plugin says about a file that holds one — silence today, the warnings the rules have to give tomorrow. Worth doing all the same: the util answers what the syntax is, and the answer should be the syntax's own.
